core/adaptive/config.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 1999-2020 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adaptive
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/alibaba/sentinel-golang/util"
+)
+
+// AdaptiveType indicates the adaptive type.
+type AdaptiveType int32
+
+const (
+	Memory AdaptiveType = iota
+)
+
+type CalculateStrategy int32
+
+const (
+	Linear CalculateStrategy = iota
+)
+
+func (c CalculateStrategy) String() string {
+	switch c {
+	case Linear:
+		return "Linear"
+	default:
+		return "Undefined"
+	}
+}
+
+func (a AdaptiveType) String() string {
+	switch a {
+	case Memory:
+		return "Memory"
+	default:
+		return "Undefined"
+	}
+}
+
+type Config struct {
+	// ID is the unique id
+	ID string `json:"id,omitempty"`
+	// user-defined name to uniquely determine adaptive strategy
+	AdaptiveConfigName string            `json:"adaptiveConfigName"`
```

Comment:
package has the adaptive, maybe "ConfigName" is better

Reply:
done